config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml

```
@@ -474,7 +474,7 @@ presubmits:
         - --gcp-region=us-central1
         - --provider=gce
         - --runtime-config=api/all=true
-        - --test_args=--ginkgo.focus=\[Feature:(WatchList|InPlacePodVerticalScaling|APIServerTracing|SidecarContainers|StorageVersionAPI|PodPreset|ClusterTrustBundle|ClusterTrustBundleProjection|PodLifecycleSleepAction|RecoverVolumeExpansionFailure)\] --ginkgo.skip=\[Feature:(SCTPConnectivity|Volumes|Networking-Performance)\]|IPv6|csi-hostpath-v0|\[KubeUp\] --minStartupPods=8
+        - --test_args=--ginkgo.focus=\[Feature:(Alpha|WatchList|InPlacePodVerticalScaling|APIServerTracing|SidecarContainers|StorageVersionAPI|PodPreset|ClusterTrustBundle|ClusterTrustBundleProjection|PodLifecycleSleepAction|RecoverVolumeExpansionFailure)\] --ginkgo.skip=\[Feature:(Beta|SCTPConnectivity|Volumes|Networking-Performance)\]|IPv6|csi-hostpath-v0|\[KubeUp\] --minStartupPods=8
```

Instead of tweaking focus regexps further I prefer to convert to convert to `-ginkgo.label-filter`. Let's end the pain.

---

in this case, would that mean the following?
```suggestion
        - --test_args=--ginkgo.label-filter=/Feature:(Alpha|WatchList|InPlacePodVerticalScaling|APIServerTracing|SidecarContainers|StorageVersionAPI|PodPreset|ClusterTrustBundle|ClusterTrustBundleProjection|PodLifecycleSleepAction|RecoverVolumeExpansionFailure)/ && !(/Feature:(Beta|SCTPConnectivity|Volumes|Networking-Performance)/ || IPv6 || csi-hostpath-v0 || KubeUp) --minStartupPods=8
```

Though I am not sure what "IPv6" and "csi-hostpath-v0" stand for, these don't appear to be labels in the k/k e2e test suite. I actually couldn't find mention of "csi-hostpath-v0" in the whole k/k codebase.

---

No, that's not quite how Ginkgo --label-filter works. See https://github.com/kubernetes/test-infra/issues/32911 and https://groups.google.com/a/kubernetes.io/g/dev/c/DTFEng143NY/m/FqdD3RADAQAJ for instructions.

A good first step would be to explain in plain English which tests should run in this job. Is it generic (= run everything that can run) or specific to certain features (as implied by the current focus expression which only runs tests selected by that expression)?

---

That is actually quite a good mental exercise. I wonder what this and the below tests are supposed to do. From both the test name and the current test definition, I suspect the plain English explanation would be `run tests for these specific features as long as they are alpha`.

---

to unblock ClusterTrustBundle beta promotion, if something like https://github.com/kubernetes/kubernetes/pull/130580 or https://github.com/kubernetes/kubernetes/pull/130567 can get us coverage via the kind alpha / alpha-beta / beta jobs (which it seems like they both can), I'd actually be fine just dropping ClusterTrustBundle|ClusterTrustBundleProjection from ~all the other e2e job configurations for now

---

> That is the million dollar question. The answer might very well be "generic job which supports all feature-gated tests that have no special other requirements", in which case the label filter is as simple as the one in [the kind jobs](https://github.com/kubernetes/test-infra/blob/e6fe78624c114fe78fc8aafe79b1b003cd4f5a76/config/jobs/kubernetes/sig-testing/kubernetes-kind.yaml#L318-L319).

IMHO that should be the end-state.

> Most of the focus expression then becomes historic garbage that we should get rid of.

+1

> Who owns these jobs? The job owner should know what they are for and what they are willing to support with regards to triaging test failures.

The GCE ones are tragedy of the commons.
The kind ones @aojea and I are actively maintaining.

---

> > in which case the label filter is as simple as the one in [the kind jobs](https://github.com/kubernetes/test-infra/blob/e6fe78624c114fe78fc8aafe79b1b003cd4f5a76/config/jobs/kubernetes/sig-testing/kubernetes-kind.yaml#L318-L319).
>
> IMHO that should be the end-state.

Cross-reference: https://github.com/kubernetes/kubernetes/pull/130580#discussion_r1982790680 proposes a slightly different label filter for "cluster with default configuration".

> The GCE ones are tragedy of the commons.

Should we make that official and move them under SIG Testing or Release? SIG release might be more appropriate, considering that they already get the alerts ([`testgrid-alert-email: release-team@kubernetes.io`](https://github.com/kubernetes/test-infra/blob/9b9d557dc54dea62e0919222c1ad10766fe4da5e/config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml#L825C5-L825C53)).